vortex-array/src/builders/mod.rs

```diff
@@ -208,15 +208,17 @@ pub trait ArrayBuilderExt: ArrayBuilder {
                 })
             }
             DType::Decimal(..) => {
+                let self_dtype = self.dtype().clone();
                 macro_rules! append_decimal {
                     ($self:expr, $ty:ty, $value:expr) => {{
                         $self
                             .as_any_mut()
                             .downcast_mut::<DecimalBuilder<$ty>>()
                             .ok_or_else(|| {
                                 ::vortex_error::vortex_err!(
-                                    "Cannot append decimal scalar of type {} to builder of another type",
+                                    "Cannot append decimal scalar of type {} to builder of type {}",
```

Comment:
Inline self_dtype here

Reply:
I'm going to remove it, I'm using this branch to iterate on the Azure Iceberg thing. It turns out there are quite a few broken things with Decimals :P

Reply:
Converted the branch back to draft to clarify the intention, once I finish fixing the latent issues will promote to normal PR again